chainer/functions/pooling/unpooling_2d.py

```diff
@@ -0,0 +1,57 @@
+import numpy
+
+from chainer import cuda
+from chainer.functions.pooling import pooling_2d
+from chainer.utils import conv
+
+
+class Unpooling2D(pooling_2d.Pooling2D):
+
+    """Unpooling over a set of 2d planes."""
+
+    def __init__(self, ksize, stride=None, pad=0,
+                 outsize=None, cover_all=True, use_cudnn=True):
+        super(Unpooling2D, self).__init__(
+            ksize, stride, pad, cover_all, use_cudnn)
```

Comment:
`Pooling2D.__init__` just sets `cover_all` and `use_cudnn` as attirbutes. So they do not affect the behavior of this function for now.
For `use_cudnn`, we do not have a way to utilize cuDNN for implementing `Unpooling`. So I think just setting `use_cudnn` to `False` and removing it from constructor are enough.
I think it is beneficial that `Unpooling` supports `cover_all` mode. To do that, we need to make `get_deconv_outsize` support `cover_all` mode and feed `cover_all` option to `get_conv_outsize` in backward. But it is impossible to add `cover_all` argument to `get_deconv_outsize` because of API compatibility policy. So we should create another version of `get_deconv_outsize` function that support `cover_all` option.